nebula2/gclient/net/Session.py

```diff
@@ -39,24 +39,97 @@ def execute(self, stmt):
             resp = self._connection.execute(self._session_id, stmt)
             end_time = time.time()
             return ResultSet(resp,
-                             all_latency=int((end_time - start_time) * 1000000),
+                             all_latency=int(
+                                 (end_time - start_time) * 1000000),
                              timezone_offset=self._timezone_offset)
         except IOErrorException as ie:
             if ie.type == IOErrorException.E_CONNECT_BROKEN:
                 self._pool.update_servers_status()
                 if self._retry_connect:
                     if not self._reconnect():
                         logging.warning('Retry connect failed')
-                        raise IOErrorException(IOErrorException.E_ALL_BROKEN, ie.message)
+                        raise IOErrorException(
+                            IOErrorException.E_ALL_BROKEN, ie.message)
                     resp = self._connection.execute(self._session_id, stmt)
                     end_time = time.time()
                     return ResultSet(resp,
-                                     all_latency=int((end_time - start_time) * 1000000),
+                                     all_latency=int(
+                                         (end_time - start_time) * 1000000),
                                      timezone_offset=self._timezone_offset)
             raise
         except Exception:
             raise
 
+    def execute_json(self, stmt):
+        """execute statement and return the result as a JSON string
+            Date and Datetime will be returned in UTC
+            JSON struct:
+              "results": [
+               {
+                  "columns": [],
+                  "data": [
+                    {
+                       "row": [ row-data ],
+                       "meta": [ metadata ]
+                    },
+                  ],
+                  "latencyInUs" : 0,
+                  "spaceName": "",
+                  "planDesc ": {
+                    "planNodeDescs": [ {
+                      "name" : "",
+                      "id" : 0,
+                      "outputVar" : "",
+                      "description" : {"key" : ""},
+                      "profiles" : [{
+                        "rows" : 1,
+                        "execDurationInUs" : 0,
+                        "totalDurationInUs" : 0,
+                        "otherStats" : {}, // map
+                      }],
+                      "branchInfo" : {
+                        "isDoBranch" : false,
+                        "conditionNodeId" : -1,
+                      },
+                      "dependencies" : [] // vector of ints
+                      }
+                    ],
+                    "nodeIndexMap" : {},
+                    "format" : "",
+                    "optimize_time_in_us" : 0,
+                  },
```

Comment:
Are all messages useful for users(such as `planDesc`)? if not, is it better to simply the json content returned to users?

Reply:
`planDesc` will only be filled when the user executes a query with `PROFILE/EXPLAIN`.
This JSON format is compatible with the neo4j format.